src/generic_ui/polymer/root.ts

```diff
@@ -90,17 +92,60 @@ Polymer({
     // to sync the value to core
     core.updateGlobalSettings(model.globalSettings);
   },
-  showToast: function(e, data) {
-    this.$.toast.setAttribute('text', data.text);
-    this.$.toast.show();
-  },
   signalToFireChanged: function() {
     if (this.ui.signalToFire != '') {
       this.fire('core-signal', {name: this.ui.signalToFire});
       this.ui.signalToFire = '';
     }
   },
-  observe: {
-    'ui.signalToFire' : 'signalToFireChanged'
+  /* All functions below help manage paper-toast behaviour. */
+  showToast: function(e, data) {
```

Comment:
Can we remove showToast and the "show-toast" signal, and just set this.ui.toastMessage?